src/Dhall/Import.hs

```diff
@@ -718,10 +737,87 @@ loadStaticWith from_import ctx n (Embed import_) = do
                 else throwM (Imported (import_:imports) (HashMismatch {..}))
 
     return expr
-loadStaticWith from_import ctx n expr = fmap join (traverse process expr)
+loadStaticWith from_import ctx n expr = fmap join (importMapM process expr)
   where
     process import_ = loadStaticWith from_import ctx n (Embed import_)
 
+{-| This is basically just a monomorphic implementation of @traverse@
+    for @Expr@, with the exception for the @ImportAlt@ constructor, that
+    gets eliminated while we search for imports that fail to resolve.
+-}
+importMapM
```

Comment:
Since this function seems specialized to `loadStaticWith`, I'd suggest inlining it in `loadStaticWith`, like this:

```haskell
loadStaticWith from_import ctx n expr = case expr of
    ImportAlt a b -> ...
    Const a -> ...
    Var a -> ...
```